### PR TITLE
[TEVA-1816] Trigger events for vacancy page interactions

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1816

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-2866.london.cloudapps.digital>

File: spec/requests/jobseekers/saved_jobs_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe "Jobseeker saves jobs", type: :request do
+  let(:school) { create(:school) }
+  let(:vacancy) { create(:vacancy) }
+  let(:credentials) { { email: "test@example.com", password: "password" } }
+  let(:jobseeker) { Jobseeker.create(credentials) }
```
Comment (cpjmcquillan) [inline spec/requests/jobseekers/saved_jobs_spec.rb:7]: We have a jobseeker factory

File: spec/requests/subscriptions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+
+RSpec.describe "Jobseeker saves jobs", type: :request do
+  let(:school) { create(:school) }
+  let(:vacancy) { create(:vacancy) }
+  let(:origin) { "/jobs/#{vacancy.id}/#{vacancy.slug}" }
+
+  before do
+    vacancy.organisation_vacancies.create(organisation: school)
```
Comment (cpjmcquillan) [inline spec/requests/subscriptions_spec.rb:9]: Do you prefer this way to setting it in the factory (e.g `organisation_vacancies_attributes: ...`)

Comment (cesidio) [reply spec/requests/subscriptions_spec.rb]: organisation_vacancies_attributes is more minimalist, let's use it!

File: spec/requests/jobseekers/job_applications_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+RSpec.describe "Job applications", type: :request do
+  let(:vacancy) { create(:vacancy, organisation_vacancies_attributes: [{ organisation: build(:school) }]) }
+  let(:jobseeker) { create(:jobseeker) }
+
+  before do
+    allow(JobseekerApplicationsFeature).to receive(:enabled?).and_return(true)
+    sign_in jobseeker
```
Comment (cpjmcquillan) [inline spec/requests/jobseekers/job_applications_spec.rb:9]: Where does this come from? In their docs I only ever saw `login_as(jobseeker, scope: :jobseeker)`

**Review by cpjmcquillan — APPROVED** [review]
Only a small question about how we sign in jobseekers in specs

File: app/controllers/subscriptions_controller.rb [hunk 4]
```diff
@@ -75,6 +75,17 @@ def destroy
 
   private
 
+  def track_origin_and_trigger_event
+    return unless origin_param&.start_with?(%r{/\w})
```
Comment (david-mears-dfe) [inline app/controllers/subscriptions_controller.rb:79]: What does the forward slash `/` do?

Is the purpose here to only track the origin if there is an origin?

File: spec/requests/jobseekers/job_applications_spec.rb (hunk 3 above)
Comment (cesidio) [reply spec/requests/jobseekers/job_applications_spec.rb]: https://github.com/heartcombo/devise/wiki/How-To:-sign-in-and-out-a-user-in-Request-type-specs-(specs-tagged-with-type:-:request)

File: app/controllers/subscriptions_controller.rb (hunk 4 above)
Comment (cesidio) [reply app/controllers/subscriptions_controller.rb:79]: yes I believe so, I just moved this logic, did not change it

**Comment by cesidio** [comment]
> Only a small question about how we sign in jobseekers in specs

sign_in(jobseeker, scope: :jobseeker)

**Pull request merged**